sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java

```diff
@@ -33,16 +38,26 @@
  * to eliminate the method definition differences.
  */
 class ConsumerSpEL {
-  SpelParserConfiguration config = new SpelParserConfiguration(true, true);
-  ExpressionParser parser = new SpelExpressionParser(config);
+  private static final Logger LOG = LoggerFactory.getLogger(ConsumerSpEL.class);
 
-  Expression seek2endExpression =
+  private SpelParserConfiguration config = new SpelParserConfiguration(true, true);
+  private ExpressionParser parser = new SpelExpressionParser(config);
+
+  private Expression seek2endExpression =
       parser.parseExpression("#consumer.seekToEnd(#tp)");
 
-  Expression assignExpression =
+  private Expression assignExpression =
       parser.parseExpression("#consumer.assign(#tp)");
 
-  public ConsumerSpEL() {}
+  private Method timestampMethod;
+
+  public ConsumerSpEL() {
+    try {
+      timestampMethod = ConsumerRecord.class.getMethod("timestamp", (Class<?>[]) null);
```

Comment:
only client 0.10+, server 0.10+ has this method, use this to detect the client version, same for server as Kafka client is not backward compatible.